### DISCUSSION TITLE
Spawn different SpriteBundles for different states

### DISCUSSION THREAD

**Discussion by gmoller** [bevyengine/bevy#2909] [Q&A] [answered]
Hi,

I've followed the example in https://bevy-cheatbook.github.io/programming/states.html but can not figure out how to spawn different SpriteBundles for different states. Is there a way to do this? The only thing I can think of is to despawn them all and re-spawn new ones on the change of state.

Thanks,
Greg

**Comment by payload** [comment, marked as answer]
I think this is a completely valid approach to despawn all sprites and spawn new ones upon state change. This provides most flexibility between states, for example in vastly different numbers of sprites completely not related to the entities in the state before.

What is your use case?

**Reply by gmoller** [reply]
Ok, makes sense. I'm trying to write a turn-based strategy type game where I have two main "states": a strategical (world map) for most of the gameplay and a more tactical one for the battles. So when transitioning to the tactical "battle stage", I figure I should add the various artifacts and remove them when returning back to the strategical state (perhaps attaching a specific marker so they can be easily queried). I think I will probably just leave the main strategical state entities as they are (just not applying the systems wired up for that state).

Thanks for the reply. I'm having a lot of fun learning Rust and Bevy by the way :). Or maybe I should say trying to learn, it's quite a change from C# which I'm used to. But a nice change.
